README.md

```diff
@@ -48,30 +48,31 @@ Because you are viewing a protected topic, you must provide `api_key` and `api_u
 Here is an example of an implementation:
 
 ```python
-engage_path = "/engage"
-engage_docs = EngagePages(
-    parser=EngageParser(
-        api=DiscourseAPI(
-            base_url="https://discourse.ubuntu.com/",
-            session=session,
-            api_key="secretkey", # API KEY used in the tests
-            api_username="canonical",
-        ),
-        index_topic_id=17229,
-        url_prefix=engage_path,
+engage_pages = EngagePages(
+    api=DiscourseAPI(
+        base_url="https://discourse.ubuntu.com/",
+        session=session,
+        get_topics_query_id=14,
+        api_key=DISCOURSE_API_KEY, # replace with your API key
+        api_username=DISCOURSE_API_USERNAME, # replace with correspoding username
     ),
-    document_template="/engage/base.html",
-    url_prefix=engage_path,
-    blueprint_name="engage-pages",
+    category_id=51,
+    page_type="engage-pages", # one of ["engage-pages", "takeovers"]
+    url_prefix="/engage",
+    skip_posts=[] # this is a list of topic ids that we want to exclude from Markdown error checks
 )
 ```
 
-Additionally, if you need a list of all engage pages, you would construct a view this way:
+In your project, you need to create your own views:
```

Comment:
Was there a reason that you decided to put the view definition in the application rather than the module? I don't necessarily disagree, but it's not what we've done with the docs stuff is it - doesn't the module define those views? Also, it feels a little odd to be requiring `url_prefix` above but then making them declare it again in the view definition.

Reply:
Essentially because we often get requests to do things in the individual engage page template (things like showing blogs on an individual engage page or show an individual engage page on a specific page are not possible now), this will give us more flexibility to potentially pass arguments to the view without doing workarounds.

In fact, the `takeovers.json` endpoint is kind of already doing this, it's an individual view and we consume the content using JS. We could potentially do this also for engage pages, because we are getting more and more requests to put engage pages in a number of places (canonical v2, which eventually didn't make it, required different design, different styles for engage pages, so need to manipulate the view template), that's why I think we should just pass the data.

This also includes the engage pages thank-you pages, which also require data that comes from the individual engage pages, so potentially we could refactor and even improve performance. Mainly because, up to this point, we used the `.parse()` method which is a big function that parses markdown with Beautiful soup independently from what is being requested. In this PR, I just basically split it to only the necessary data i.e. if we need an individual engage page - pass an individual engage page, if we need the whole list of engage pages - we pass that whole list.

Reply:
About the `url_prefix`, yeah I guess we can remove that requirement now.

Reply:
Yep sounds good. Thanks for the explanation.